my-react-study-app/src/widgets/itemsSection/itemsSection.tsx

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable react/jsx-no-useless-fragment */
+
+import './itemsSection.css';
+import { useEffect } from 'react';
+import { useAppDispatch, useAppSelector } from '../../shared/store/store';
+import { itemsApi } from '../../shared/api/itemsApi';
+import { ElementRequest, GetCharactersParams, ItemsSectionProps } from '../../shared/types';
+import Pagination from '../pagination/pagination';
+import Loader from '../loader/loader';
+import { itemsSlice } from '../../shared/store/itemsSlice';
+import DetailedItem from '../detailedItem/detailedItem';
+import { itemsSelectedSlice } from '../../shared/store/selectedItemsSlice';
+import DetailedItemsControls from '../detailedSection/detailedItemsControls';
+import { useTheme } from '../../shared/context/themeMode';
+
+export default function ItemsSection(props: ItemsSectionProps) {
+    const { isDark } = useTheme();
+    const { currentRequest, currentPage, currentId, section, loading } = useAppSelector((state) => state.itemsReducer);
+    const { selectedItems } = useAppSelector((state) => state.itemsSelectedReducer);
+    const { setSelectedItems, unsetSelectedItems } = itemsSelectedSlice.actions;
+    const { setLoading, setCurrentId } = itemsSlice.actions;
+    const dispatch = useAppDispatch();
+    const { data } = props;
+
+    useEffect(() => {
+        if (data) {
+            dispatch(setLoading(false));
+        }
+    }, [data]);
+
+    return (
+        <>
+            {loading ? (
+                <Loader />
+            ) : (
+                <main className={isDark ? 'main dark' : 'main light'}>
+                    <div className="main__wrapper">
+                        <div className="main__content">
+                            <div className="main__search_section">
+                                <div className="main__items">
+                                    {Object.keys(data!.info).length !== 0 && <Pagination info={data!.info} />}
```

Comment:
it's dangerous to use !. operator, try avoid use it. It's better to add additional check